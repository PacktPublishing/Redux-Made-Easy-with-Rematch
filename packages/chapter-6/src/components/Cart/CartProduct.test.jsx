import React from "react";
import { render } from "@testing-library/react";
import { CartProduct } from ".";

const productJson = {
  id: "b590e450-1e0c-4344-92b8-e1f6cc260587",
  image_url:
    "https://dummyimage.com/400x400/adca70/000&text=Fantastic Wooden Sausages",
  stock: 0,
  productName: "Fantastic Wooden Sausages",
  price: 1,
  productDescription:
    "Ea qui et. Minus minima omnis. Nisi temporibus quam vel doloribus tempore ut possimus repellendus minus. Aut et repudiandae et dolorum ut consectetur. Qui laborum tenetur corporis odit architecto. Rerum neque odit.",
  favorite: false,
};

describe("CartProduct", () => {
  it("should render the cart product correctly", () => {
    const { getByText, getByRole, getByLabelText, container } = render(
      <CartProduct product={productJson} quantity={1} />
    );

    expect(getByText(productJson.productName)).toBeInTheDocument();
    expect(getByRole("img")).toBeInTheDocument();
    expect(getByText("$1.00")).toBeInTheDocument();
    expect(getByLabelText("remove")).toBeInTheDocument();
    expect(getByLabelText("purchase more")).toBeInTheDocument();
    expect(getByLabelText("product quantity")).toBeInTheDocument();
    expect(getByLabelText("product quantity")).toContainHTML("1");
    expect(getByLabelText("purchase more")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should render the quantity correctly", () => {
    const { getByLabelText } = render(
      <CartProduct product={productJson} quantity={1423} />
    );
    expect(getByLabelText("product quantity")).toContainHTML("1423");
  });
});
