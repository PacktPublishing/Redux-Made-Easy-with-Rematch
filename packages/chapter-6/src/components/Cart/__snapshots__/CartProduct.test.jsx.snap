// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartProduct should render the cart product correctly 1`] = `
<div>
  <article
    class="p-4 flex space-x-4"
    role="listitem"
  >
    <img
      alt="Product"
      class="flex-none w-auto max-w-16 max-h-16 rounded-lg object-cover bg-gray-100"
      loading="lazy"
      src="https://dummyimage.com/400x400/adca70/000&text=Fantastic Wooden Sausages"
    />
    <div
      class="flex-auto"
    >
      <h2
        class="text-lg font-semibold text-black under"
      >
        Fantastic Wooden Sausages
      </h2>
      <p
        class="text-sm font-medium text-gray-500"
      >
        $1.00
      </p>
    </div>
    <div
      class="flex items-center"
    >
      <button
        aria-label="remove"
        class="w-5 h-5 rounded-md text-gray-400 border border-gray-300 mr-2"
        type="button"
      >
        <svg
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 12H6"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
      <div
        aria-label="product quantity"
        class="w-auto p-2 h-7 text-sm flex items-center justify-center rounded-md text-gray-500 border border-gray-300 mr-2"
      >
        1
      </div>
      <button
        aria-label="purchase more"
        class="w-5 h-5 rounded-md text-gray-400 border border-gray-300 mr-2"
        type="button"
      >
        <svg
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v6m0 0v6m0-6h6m-6 0H6"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </article>
</div>
`;
